c10/util/Half.h

```diff
@@ -33,18 +33,32 @@
 
 #ifdef __CUDACC__
 #include <cuda_fp16.h>
+#define __NOT_CPU__
```

Comment:
Since this is a header, might be better to use a macro that has TORCH or C10 in it to make it less likely to clobber someone else